pallets/staking/src/testing_utils.rs

```diff
@@ -20,105 +20,93 @@
+    let _ = T::Balances::make_free_balance_be(&c.account(), balance.into());
+    T::Currency::issue(balance.into());
+    Ok((s, c))
 }
 
 /// Create a stash and controller pair.
+/// Both accounts are created with the given balance and with DID.
 pub fn create_stash_controller<T: Trait>(
     n: u32,
-    balance_factor: u32,
-) -> Result<(T::AccountId, T::AccountId), &'static str> {
-    let amount = T::Currency::minimum_balance() * (balance_factor / 10).max(1).into();
-    create_stash_controller_with_balance::<T>(n, amount)
+    balance: u32,
+) -> Result<(User<T>, User<T>), DispatchError> {
+    _create_stash_controller::<T>(n, balance, RewardDestination::Staked, false)
 }
 
-/// Create a stash and controller pair.
-/// Both accounts are created with the given balance and with DID.
-pub fn create_stash_controller_with_balance<T: Trait>(
+pub fn create_stash_with_dead_controller<T: Trait>(
     n: u32,
-    balance: BalanceOf<T>,
-) -> Result<(T::AccountId, T::AccountId), &'static str> {
-    let stash = create_funded_user_with_did::<T>("stash", n, balance);
-    let controller = create_funded_user_with_did::<T>("controller", n, balance);
-    let controller_lookup: <T::Lookup as StaticLookup>::Source =
-        T::Lookup::unlookup(controller.clone());
-    let reward_destination = RewardDestination::Staked;
-    Staking::<T>::bond(
-        RawOrigin::Signed(stash.clone()).into(),
-        controller_lookup,
-        balance.into(),
-        reward_destination,
-    )?;
-    return Ok((stash, controller));
+    balance: u32,
+) -> Result<(User<T>, User<T>), DispatchError> {
+    _create_stash_controller::<T>(n, balance, RewardDestination::Controller, true)
 }
 
-/// Create a stash and controller pair, where the controller is dead, and payouts go to controller.
-/// This is used to test worst case payout scenarios.
-pub fn create_stash_and_dead_controller<T: Trait>(
+fn _create_stash_controller<T: Trait>(
     n: u32,
-    balance_factor: u32,
-) -> Result<(T::AccountId, T::AccountId), &'static str> {
-    let stash = create_funded_user::<T>("stash", n, balance_factor);
-    // controller has no funds
-    let controller = create_funded_user::<T>("controller", n, 0);
-    let controller_lookup: <T::Lookup as StaticLookup>::Source =
-        T::Lookup::unlookup(controller.clone());
-    let reward_destination = RewardDestination::Controller;
-    let amount = T::Currency::minimum_balance() * (balance_factor / 10).max(1).into();
+    balance: u32,
+    reward_destination: RewardDestination<T::AccountId>,
+    dead: bool,
+) -> Result<(User<T>, User<T>), DispatchError> {
+    let stash = create_funded_user::<T>("stash", n, balance);
+    let controller = if dead {
+        let acc: T::AccountId = account("controller", n, 100);
+        User {
+            account: acc.clone(),
+            origin: RawOrigin::Signed(acc),
+            did: None,
+            secret: None,
+            uid: None,
+        }
```

Comment:
I think `UserBuilder` allows you to do that:
```suggestion
        let acc: T::AccountId = account("controller", n, 100);
        UserBuilder::<T>::default().account(acc).build()
```

Reply:
Yeah it does but the problem is `build()` here. We want to test the condition where the controller is dead means it doesn't even have 0 balance (i.e no storage read and write before).